recipes/aws-sdk-cpp/all/conanfile.py

```diff
@@ -398,6 +402,65 @@ def build(self):
         cmake = self._configure_cmake()
         cmake.build()
 
+    @property
+    def _res_folder(self):
+        return "res"
+
+    def _create_project_cmake_module(self):
+        # package files needed to build other components (e.g. aws-cdi-sdk) with this SDK
+        for file in [
+            "cmake/compiler_settings.cmake",
+            "cmake/initialize_project_version.cmake",
+            "cmake/utilities.cmake",
+            "toolchains/cmakeProjectConfig.cmake",
+            "toolchains/pkg-config.pc.in",
+            "aws-cpp-sdk-core/include/aws/core/VersionConfig.h"
+        ]:
+            self.copy(file, src=self._source_subfolder, dst=self._res_folder)
+            tools.replace_in_file(os.path.join(self.package_folder, self._res_folder, file), "CMAKE_CURRENT_SOURCE_DIR", "AWS_NATIVE_SDK_ROOT", strict=False)
+
+        # avoid getting error from hook
+        with tools.chdir(os.path.join(self.package_folder, self._res_folder)):
+            rename(self, os.path.join("toolchains", "cmakeProjectConfig.cmake"), os.path.join("toolchains", "cmakeProjectConf.cmake"))
+            tools.replace_in_file(os.path.join("cmake", "utilities.cmake"), "cmakeProjectConfig.cmake", "cmakeProjectConf.cmake")
+
+        # create a cmake module to load the files above
+        contents = textwrap.dedent("""
```

Comment:
I think you copied this from `cmake/AWSSDKConfig.cmake`?
I believe this contains potential version depdendent logic.
Can we move this to a patch, and copy that file here instead?

Reply:
I tried to have the minimum amount of CMake code required to build a single SDK.  
it's copied from different parts of the CMake code. That specific file is actually not seemed to be used anymore by the package.

Reply:
I'm not sure how to achieve that with a patch

Reply:
The following patch creates a file containing 3 lines:
```patch
--- /dev/null
+++ some_file
@@ -0,0 +1,3 @@
+line1
+line2
+line3
```
Using the same format, you can create a patch for a longer file.